src/core/metrics/metrics_util.go

```diff
@@ -357,3 +376,29 @@ func GetCalculationMap() map[string]string {
 		"container.mem.oom.kill":                             "avg",
 	}
 }
+
+func GenerateMetricsReports(entities []*StatsEntityWrapper) (bundles []core.Payload) {
```

Comment:
I think you can change the return type here to `bundle core.Payload`, since we always return an array with one payload in it.

Also a more descriptive name like `GenerateMetricsReportBundle` might be better.

Reply:
done.